DogServiceExample/src/main/java/com/common/app/Dog.java

```diff
@@ -0,0 +1,102 @@
+package com.common.app;
+
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import com.common.rest.DogService;
+
+/**
+ * Created by adyachenko on 7/22/2016.!
+ */
+public class Dog {
+    private int id;
+    private String name;
+    private String dateOfBirth;
+    private int height;
+    private int weight;
+
+    public Dog() {
+        id = -1;
+        name = null;
+        dateOfBirth = null;
+        height = -1;
+        weight = -1;
+        DogService.incrementDogId();
```

Comment:
This introduces two way dependency, which makes application more comples.
Ideally your Dog should know nothing about DogService.